samples/nrf9160/nrf_cloud_mqtt_multi_service/src/connection.c

```diff
@@ -274,6 +284,24 @@ static void cloud_event_handler(const struct nrf_cloud_evt *nrf_cloud_evt)
 	case NRF_CLOUD_EVT_RX_DATA_GENERAL:
 		LOG_DBG("NRF_CLOUD_EVT_RX_DATA_GENERAL");
 		LOG_DBG("%d bytes received from cloud", nrf_cloud_evt->data.len);
+
+		/* Pass the device message along to the application, if it is listening */
+		if (general_dev_msg_handler) {
+			/* To keep the sample simple, we invoke the callback directly.
+			 * If you want to do complex operations in this callback without blocking
+			 * receipt of data from nRF Cloud, you should set up a work queue and pass
+			 * messages to it either here, or from inside the callback.
+			 *
+			 * Again, for simplicity, we assume that the device message payload does
+			 * not contain any NULL bytes. If you expect to receive device messages
+			 * that may contain NULL bytes, you will need to adjust this logic.
```

Comment:
what is the scenario that would produce binary data to come in on this event?
and if it did, the msg handling code deals with it safely, right?

Reply:
> what is the scenario that would produce binary data to come in on this event?

I thought we allowed customers to send whatever they want on c2d, though I can't remember if that was via cloud to cloud, or by directly connecting to the MQTT network. Is that more restricted than I thought?

Reply:
🤔 i don't recall, a good question to ask the backend team.
regardless, receiving a binary payload seems to be handled ok by your code.

Reply:
Good point -- That JSON parse attempt is robust against binary data, so long as it is NULL-terminated (and it is!). When I first wrote the function I had some extra validity checking that, ironically, was not robust against it.